samples/durable-task-sdks/dotnet/Agents/PromptChaining/Worker/Services/AgentServices.cs

```diff
@@ -0,0 +1,433 @@
+using Microsoft.Extensions.Logging;
+using AgentChainingSample.Shared.Models;
+using Azure.Identity;
+using Azure.Core;
+
+namespace AgentChainingSample.Services;
+
+/// <summary>
+/// Research agent service for news article research
+/// </summary>
+public class ResearchAgentService : BaseAgentService
+{
+    private const string AgentName = "ResearchAgent";
+    private const string EndpointEnvVar = "AGENT_CONNECTION_STRING";
+    private readonly string _systemPrompt = @"You are an expert research agent specializing in news topics.
+Your task is to analyze topics, identify key facts, and organize information in a journalistic format.
+Focus on accuracy, thoroughness, and objectivity when researching any topic.";
+    
+    private bool _initialized = false;
+
+    public ResearchAgentService(ILogger<ResearchAgentService> logger) 
+        : base(Environment.GetEnvironmentVariable(EndpointEnvVar) ?? 
```

Comment:
Rather than using `Environment.GetEnvironmentVariable`, you should use an injected `IConfiguration` service to look up environment variables. This follows best practices for .NET Core apps.